src/compiler/checker.ts

```diff
@@ -22628,11 +22628,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             let i = 0;
             for (const type of target.types) {
                 const targetType = getTypeOfPropertyOfType(type, propertyName);
-                if (targetType && related(getDiscriminatingType(), targetType)) {
-                    discriminable[i] = discriminable[i] === undefined ? true : discriminable[i];
-                }
-                else {
-                    discriminable[i] = false;
+                if (discriminable[i] === undefined) {

```

Comment:
To explain this change: [This line](https://insiders.vscode.dev/github/MariaSolOs/TypeScript/blob/86295fb3d7ffa184f2f00f539171c56c56826961/src/compiler/checker.ts#L22621-L22622) says that "discriminated types stop at 'true'", but the code is overwriting a potentially true `discriminable[i]` to `false` if one of the union variants has a property with a type that's unrelated to the target type.